<?xml version='1.0'?>
<ns2:test-suite xmlns:ns2='urn:model.allure.qatools.yandex.ru' start='1748356968541' stop='1748357104304'>
    <name>[SMOKE] Get server recommendations endpoint</name>
    <title>[SMOKE] Get server recommendations endpoint</title>
    <test-cases>
        <test-case start='1748356968543' status='failed' stop='1748356983609'>
            <name>should respond with valid status code and object fields</name>
            <title>should respond with valid status code and object fields</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: get</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:11:26)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748356983611' status='failed' stop='1748356998685'>
            <name>should respond with valid status code and proper data fields when using coordinates filter</name>
            <title>should respond with valid status code and proper data fields when using coordinates filter</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: get</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:222:26)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748356998688' status='failed' stop='1748357013829'>
            <name>should respond with valid status code, proper data fields and limited count of servers when using limits filter</name>
            <title>should respond with valid status code, proper data fields and limited count of servers when using limits filter</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: get</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:436:26)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357013831' status='failed' stop='1748357028905'>
            <name>put method should respond with status code 405 and an error message</name>
            <title>put method should respond with status code 405 and an error message</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: put</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:649:28)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357028907' status='failed' stop='1748357043970'>
            <name>post method should respond with status code 405 and an error message</name>
            <title>post method should respond with status code 405 and an error message</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: post</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:649:28)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357043972' status='failed' stop='1748357059035'>
            <name>patch method should respond with status code 405 and an error message</name>
            <title>patch method should respond with status code 405 and an error message</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: patch</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:649:28)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357059038' status='failed' stop='1748357074095'>
            <name>del method should respond with status code 405 and an error message</name>
            <title>del method should respond with status code 405 and an error message</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: delete</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:649:28)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357074096' status='failed' stop='1748357089235'>
            <name>should return United States of America as a recommendation when the coordinates are below min or above max values</name>
            <title>should return United States of America as a recommendation when the coordinates are below min or above max values</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: get</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at runNextTicks (node:internal/process/task_queues:60:5)
                    at listOnTimeout (node:internal/timers:540:9)
                    at process.processTimers (node:internal/timers:514:7)
                    at async Promise.all (index 0)
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:669:27)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1748357089245' status='failed' stop='1748357104301'>
            <name>should only return online servers</name>
            <title>should only return online servers</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>Error: Endpoint is not responding. 5 attempts has been made with 3 seconds apart. Endpoint: https://127.0.0.1/v1/servers/recommendations. Method: get</message>
                <stack-trace>    at next (/app/node_modules/@nordvpn/tests-backend-core/src/request.js:38:11)
                    at /app/node_modules/@nordvpn/tests-backend-core/src/request.js:34:16
                    at async UserContext.&lt;anonymous> (/app/tests/smoke/serverRecommendations.spec.js:691:26)</stack-trace>
            </failure>
        </test-case>
    </test-cases>
</ns2:test-suite>